projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/DeployJob.tsx

```diff
@@ -48,46 +49,77 @@ export default class DeployJobDialog extends Component<IJobFullProps> {
       </>
     );
   }
+
+  handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const eventDetail = { runBeforeDeploy: event.target.checked };
+    const customEvent = new CustomEvent('deployRunChange', {
+      detail: eventDetail
+    });
+    window.dispatchEvent(customEvent);
+  };
 }
 
 export async function showCreateDeploymentDialog() {
+  let runBeforeDeploy = false;
+
+  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    runBeforeDeploy = event.target.checked;
+  };
+
   const result = await showDialog({
     title: CREATE_DEP_BUTTON_LABEL,
     body: (
-      <DeployJobDialog
-        jobName={jobData.get(VdkOption.NAME)!}
-        jobPath={jobData.get(VdkOption.PATH)!}
-        jobTeam={jobData.get(VdkOption.TEAM)!}
-      ></DeployJobDialog>
+      <>
+        <DeployJobDialog
+          jobName={jobData.get(VdkOption.NAME) || ''}
+          jobPath={jobData.get(VdkOption.PATH) || ''}
+          jobTeam={jobData.get(VdkOption.TEAM) || ''}
+        />
+        <div>
+          <input
+            type="checkbox"
+            name="deployRun"
+            id="deployRun"
+            className="jp-vdk-checkbox"
+            onChange={handleCheckboxChange}
+          />
+          <label className="checkboxLabel" htmlFor="deployRun">
+            Run before deployment
+          </label>
+        </div>
```

Comment:
Consider further component modularization (see Separation of Concerns). The checkbox for 'Run before deployment' can be a separate reusable component.